File: backend/test/java/de/griefed/serverpackcreator/utilities/commonutilities/WebUtilitiesTest.java
package de.griefed.serverpackcreator.utilities.commonutilities;

import de.griefed.serverpackcreator.utilities.commonutilities.WebUtilities;
import org.apache.commons.io.FileUtils;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.io.File;
import java.net.URL;

public class WebUtilitiesTest {

    private final WebUtilities WEB_UTILITIES;

    WebUtilitiesTest() {
        this.WEB_UTILITIES = new WebUtilities();
    }

    @Test
    void downloadFileTest() {
        try {
            WEB_UTILITIES.downloadFile("Fabric-Server-Launcher.jar",new URL("https://meta.fabricmc.net/v2/versions/loader/1.18.1/0.12.12/0.10.2/server/jar"));
        } catch (Exception ignored) {}
        Assertions.assertTrue(new File("Fabric-Server-Launcher.jar").exists());
        FileUtils.deleteQuietly(new File("Fabric-Server-Launcher.jar"));
        try {
            WEB_UTILITIES.downloadFile("some_foooooolder/foooobar/Fabric-Server-Launcher.jar",new URL("https://meta.fabricmc.net/v2/versions/loader/1.18.1/0.12.12/0.10.2/server/jar"));
        } catch (Exception ignored) {}
        Assertions.assertTrue(new File("some_foooooolder/foooobar/Fabric-Server-Launcher.jar").exists());
        FileUtils.deleteQuietly(new File("some_foooooolder"));
    }
}

File: serverpackcreator-api/src/jvmTest/kotlin/de/griefed/serverpackcreator/api/common/WebUtilitiesTest.kt
package de.griefed.serverpackcreator.api.common

import de.griefed.serverpackcreator.api.ApiWrapper
import de.griefed.serverpackcreator.api.utilities.common.WebUtilities
import de.griefed.serverpackcreator.api.utilities.common.deleteQuietly
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.io.File
import java.net.URI

class WebUtilitiesTest internal constructor() {
    private var webUtilities: WebUtilities =
        ApiWrapper.api(File("src/jvmTest/resources/serverpackcreator.properties")).webUtilities

    @Suppress("SpellCheckingInspection")
    @Test
    fun downloadFileTest() {
        val file = File("tests/Fabric-Server-Launcher.jar")
        try {
            webUtilities.downloadFile(
                file,
                URI("https://meta.fabricmc.net/v2/versions/loader/1.18.1/0.12.12/0.10.2/server/jar").toURL()
            )
        } catch (ignored: Exception) {
        }
        Assertions.assertTrue(file.exists())
        file.deleteQuietly()
        val nested = File("tests/some_foooooolder/foooobar/Fabric-Server-Launcher.jar")
        try {
            webUtilities.downloadFile(
                nested,
                URI("https://meta.fabricmc.net/v2/versions/loader/1.18.1/0.12.12/0.10.2/server/jar").toURL()
            )
        } catch (ignored: Exception) {
        }
        Assertions.assertTrue(nested.exists())
        nested.deleteQuietly()
    }
}